rbx_binary/src/serializer.rs

```diff
@@ -443,7 +443,15 @@ impl<'a, W: Write> BinarySerializer<'a, W> {
                         } else {
                             instance
                                 .properties
-                                .get(prop_name)
+                                .get(
+                                    if let Some(descriptor) =
+                                        find_canonical_property_descriptor(type_name, prop_name)
+                                    {
+                                        &*descriptor.name
+                                    } else {
+                                        prop_name
+                                    },
```

Comment:
This is a much too complicated expression to slap inline here. I need to think over the implications of this as well.

Reply:
This is also a problem because we're redundantly looking up the same property descriptor in the database for every instance. This process is expensive and can take a long time.

Reply:
I am working on what I think is the correct solution to this problem. We should avoid introducing a fix here, especially as part of a regular property feature implementation.

Reply:
Should I have `deserializer.rs` use `Color3uint8` as the key for the property map instead of the canonical name (as described in the original PR comment)? Or perhaps check explicitly in `serializer.rs` (similar to `Name`) and do `instance.properties.get("Color")`? I'm grepping through the lua database and it looks like this type is exclusively used by `BasePart` and friends, so either way seems like it would work. Your call.

Reply:
We should do neither here, the change I'm working on should solve this problem with no hard-coded special cases. `Color` is not quite the same level of specialness as `Name` 😄